crates/protocol/protocol/src/block.rs

```diff
@@ -143,6 +149,43 @@ impl L2BlockInfo {
         Self { block_info, l1_origin, seq_num }
     }
 
+    /// Constructs an [L2BlockInfo] from a given [alloy_rpc_types_eth::Block] and [ChainGenesis].
+    pub fn from_rpc_block_and_genesis(
```

Comment:
Instead of introducing a new function specific to the rpc block, can we convert between the RPC block and the consensus block? This should work:

```rs
let consensus_block = self
    .l2_provider
    .get_block(
        BlockNumberOrTag::Safe.into(),
        BlockTransactionsKind::Full,
    )
    .await?
    .map(|b| b.into_consensus().map_transactions(|b| b.inner.inner))?;
```

Reply:
actually let me check if we have that conversion on the rpc type, if not I'll add

Reply:
Thanks @mattsse - I didn't see it so this is why i just went with the dup for now. Was going to raise this as an issue if not :)